frontend/src/pages/Q&A/Q&A.jsx

```diff
@@ -161,12 +165,117 @@ function Ques(props) {
   function DeactiveButton() {
     setButtonPressed(!isButtonPressed);
   }
+
+  const [getQuestions, setQuestions] = useState([]);
+
+  function getQues() {
+    fetch(`${END_POINT}/question/getallquestions`, {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+      },
+    })
+      .then((res) => res.json())
+      .then((data) => {
+        setLoading(false);
+        setQuestions(data);
+        // console.log(data);
+      });
+  }
+
+  const upvote = async (questionId) => {
+    const response = await fetch(`${END_POINT}/question/upvote`, {
+      method: "PATCH",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ questionId }),
+    });
+
+    if (!response.ok) {
+      throw new Error("Failed to upvote question");
+    }
+    // const data = await response.json();
+    getQues();
+    setToastMessage("Upvote Successfully");
+    setOpenToast(true);
+    setSeverity("success");
+  };
+
+  const downvote = async (questionId) => {
+    const response = await fetch(`${END_POINT}/question/downvote`, {
+      method: "PATCH",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ questionId }),
+    });
+
+    if (!response.ok) {
+      throw new Error("Failed to upvote question");
+    }
+    // const data = await response.json();
+    // console.log(data);
+    getQues();
+    setToastMessage("Downvote Successfully");
+    setOpenToast(true);
+    setSeverity("success");
+  };
+
+  useEffect(() => {
+    getQues();
+    setToastMessage("Fetching Questions...");
```

Comment:
we shouldn't be showing this toast message while fetching the data. Just loader animation while fetching is fine. Please check https://github.com/HITK-TECH-Community/Community-Website/blob/main/frontend/src/pages/ContactUs/ContactUs.jsx#L323-L328 this section for reference. For every api call, there can be success/failure and the toast should be showing success message in green if api is successful or failure message in red with error message. in your pr i don't see the failure scenario being handled. Please go through the code i shared here, it will give you the idea of toast implementation

Reply:
ok i am working on this and removing the toast message on the loading.

Reply:
@Kajol-Kumari Please Check Out Now.